app/scripts/lib/multichain-api/provider-authorize.js

```diff
@@ -113,14 +115,35 @@ export async function providerAuthorizeHandler(req, res, _next, end, hooks) {
       },
     });
 
+    const mergedScopes = mergeScopes(
+      flattenedRequiredScopes,
+      flattenedOptionalScopes,
+    );
+
+    // clear per scope middleware
+    hooks.multichainMiddlewareManager.removeAllMiddleware();
+    hooks.multichainSubscriptionManager.unsubscribeAll();
+
+    // if the eth_subscription notification is in the scope and eth_subscribe is in the methods
+    // then get the subscriptionManager going for that scope
+    Object.entries(mergedScopes).forEach(([scope, scopeObject]) => {
+      if (
+        scopeObject.notifications.includes('eth_subscription') &&
```

Comment:
this brings up something interesting, we need a middleware that verifies that the outgoing message is actually allowed to be sent to the dapp

Reply:
is the idea that perhaps a permission could be removed while a subscription is active? Otherwise shouldn't this permission check suffice?

Reply:
unsure. I guess the assumption here is that notifications cannot originate elsewhere in the pipeline. This is probably fine for now